benchmarks/Makefile.am

```diff
@@ -253,50 +252,305 @@ benchmarks_protoc_inputs_proto2_message4 =                                 \
+	sed -i "s/\#include \"datasets/\#include \"gogo\/cpp_no_group\/datasets/g" gogo/cpp_no_group/cpp_benchmark.cc
+	sed -i "s/\#include \"benchmarks.pb.h/\#include \"gogo\/cpp_no_group\/benchmarks.pb.h/g" gogo/cpp_no_group/cpp_benchmark.cc  
+	touch generate_cpp_no_group_benchmark_code
+	
+bin_PROGRAMS += cpp-no-group-benchmark
+cpp_no_group_benchmark_LDADD = $(top_srcdir)/src/libprotobuf.la $(top_srcdir)/third_party/benchmark/src/libbenchmark.a
+cpp_no_group_benchmark_SOURCES = gogo/cpp_no_group/cpp_benchmark.cc
+cpp_no_group_benchmark_CPPFLAGS = -I$(top_srcdir)/src -I$(srcdir)/gogo/cpp_no_group -I$(top_srcdir)/third_party/benchmark/include
+# Explicit deps because BUILT_SOURCES are only done before a "make all/check"
+# so a direct "make test_cpp" could fail if parallel enough.
+# See: https://www.gnu.org/software/automake/manual/html_node/Built-Sources-Example.html#Recording-Dependencies-manually
+gogo/cpp_no_group/cpp_no_group_benchmark-cpp_benchmark.$(OBJEXT): $(cpp_no_group_benchmarks_protoc_outputs) $(cpp_no_group_benchmarks_protoc_outputs_proto2) $(cpp_no_group_benchmarks_protoc_outputs_header) \
+	$(cpp_no_group_benchmarks_protoc_outputs_proto2_header) $(top_srcdir)/third_party/benchmark/src/libbenchmark.a generate_cpp_no_group_benchmark_code
+gogo/cpp_no_group/cpp_benchmark.cc: generate_cpp_no_group_benchmark_code
+nodist_cpp_no_group_benchmark_SOURCES =                                    \
+	$(cpp_no_group_benchmarks_protoc_outputs_proto2)                         \
+	$(cpp_no_group_benchmarks_protoc_outputs)                                \
+	$(cpp_no_group_benchmarks_protoc_outputs_header)                         \
+	$(cpp_no_group_benchmarks_protoc_outputs_proto2_header)
+
+cpp_no_group: cpp_no_group_protoc_middleman generate_gogo_data cpp-no-group-benchmark
+	./cpp-no-group-benchmark $(gogo_data)
+ 
+gogo_proto_middleman: protoc-gen-gogoproto
+	mkdir -p "tmp/gogo_proto"
+if USE_EXTERNAL_PROTOC
+	$(PROTOC) -I$(srcdir) -I$(top_srcdir) --plugin=protoc-gen-gogoproto --gogoproto_out=./tmp $(benchmarks_protoc_inputs) $(benchmarks_protoc_inputs_benchmark_wrapper) $(benchmarks_protoc_inputs_proto2)
+else
+	oldpwd=`pwd` && ( cd $(srcdir) && $$oldpwd/../src/protoc$(EXEEXT) -I$(srcdir) -I$(top_srcdir) --plugin=protoc-gen-gogoproto --gogoproto_out=$$oldpwd/tmp/gogo_proto $(benchmarks_protoc_inputs) $(benchmarks_protoc_inputs_benchmark_wrapper) $(benchmarks_protoc_inputs_proto2) )
+endif
+	touch gogo_proto_middleman
+
+new_data = $$(for data in $(all_data); do echo "tmp/$$data"; done | xargs)
+
+generate_gogo_data: protoc_middleman protoc_middleman2 gogo-data-scrubber
+	mkdir -p `dirname $(new_data)`
+	./gogo-data-scrubber $(all_data) $(new_data)
+	touch generate_gogo_data
+	
+make_tmp_dir_gogo:
+	mkdir -p tmp/go_no_group/benchmark_code
+	mkdir -p tmp/gogofast/benchmark_code
+	mkdir -p tmp/gogofaster/benchmark_code
+	mkdir -p tmp/gogoslick/benchmark_code
+	touch make_tmp_dir_gogo
+
+if USE_EXTERNAL_PROTOC
```

Comment:
Let's remove the use of "USE_EXTERNAL_PROTOC". It's used for cross-compile but we don't use it for benchmarks.